sklearn/linear_model/_coordinate_descent.py

```diff
@@ -709,18 +713,46 @@ def fit(self, X, y, check_input=True):
             y = check_array(y, order='F', copy=False, dtype=X.dtype.type,
                             ensure_2d=False)
 
-        # Ensure copying happens only once, don't do it again if done above
+        n_samples, n_features = X.shape
+        alpha = self.alpha
+
+        if isinstance(sample_weight, numbers.Number):
+            sample_weight = None
+        if sample_weight is not None:
+            if check_input:
+                if sparse.issparse(X):
+                    raise ValueError("Sample weights do not (yet) support "
+                                     "sparse matrices.")
```

Comment:
I see tests with sample weights and sparse matrices why do you have this?

Reply:
There is `test_enet_sample_weight_sparse`, whicht tests that `ElasticNetc` raises an error as sample weights are not supported with sparse `X`.
Secondly, there is `test_enet_sample_weight_consistency` which is decorated with `@pytest.mark.parametrize('sparseX', [False])` such that this test could easily be extended once sparse `X` are supported with sample weights.

Reply:
ok I had not paid attention to `@pytest.mark.parametrize('sparseX', [False])` We tend to add code for tests when it becomes relevant. Do you plan to this next when this is merged?

Reply:
I'd say that this is needed to close #3702 and that is is a useful feature. I'm tempted to try it in a new PR, but I'm afraid that it needs a larger code change than this one.